python/pyspark/sql/tests/test_connect_compatibility.py

```diff
@@ -17,161 +17,234 @@
 
 import unittest
 import inspect
+import functools
 
 from pyspark.testing.connectutils import should_test_connect, connect_requirement_message
 from pyspark.testing.sqlutils import ReusedSQLTestCase
 from pyspark.sql.classic.dataframe import DataFrame as ClassicDataFrame
 from pyspark.sql.classic.column import Column as ClassicColumn
 from pyspark.sql.session import SparkSession as ClassicSparkSession
+from pyspark.sql.catalog import Catalog as ClassicCatalog
 
 if should_test_connect:
     from pyspark.sql.connect.dataframe import DataFrame as ConnectDataFrame
     from pyspark.sql.connect.column import Column as ConnectColumn
     from pyspark.sql.connect.session import SparkSession as ConnectSparkSession
+    from pyspark.sql.connect.catalog import Catalog as ConnectCatalog
 
 
 class ConnectCompatibilityTestsMixin:
-    def get_public_methods(self, cls):
+    def _get_public_methods(self, cls):
```

Comment:
it doesn't need to be with `_` prefix. nested funcs are fine

Reply:
Updated. Thanks!